### PR TITLE
Add drop while overload

### LINKED ISSUE #1218: add drop while element overload
(created by Steffan [here](https://chat.stackexchange.com/transcript/message/61479479))

### PR DESCRIPTION
**Pull request by Steffan153**
Closes #1218

### CODE REVIEW

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -310,14 +310,16 @@
...
-  description: Binomial coefficient / choose a random items from b / same except duplicates
+  name: Choose / random choice / set same / drop while
+  description: Binomial coefficient / choose a random items from b / same except duplicates / drop while
   arity: 2
   overloads:
       num-num: a choose b (binomial coefficient)
       num-str: choose a random items from b
       str-num: choose b random items from a
       str-str: are the set of characters in the strings the same?
+      any-fun: remove items from the beginning of a until b(x) returns false
```
Comment (chunkybanana) [inline documents/knowledge/elements.yaml:321]: What is `x`? The whole array? The dropped item?

File: vyxal/elements.py [hunk 2]
```diff
@@ -3184,6 +3184,17 @@ def n_choose_r(lhs, rhs, ctx):
     (str, str) -> does a have the same characters as b
     """
     ts = vy_type(lhs, rhs)
+
+    @lazylist
+    def drop_while(vec, fun, ctx):
```
Comment (chunkybanana) [inline vyxal/elements.py:3189]: This should probably go in helpers.

File: documents/knowledge/elements.yaml (hunk 1 above)
Comment (cgccuser) [reply documents/knowledge/elements.yaml:321]: Each element of the array, I guess

Comment (cgccuser) [reply documents/knowledge/elements.yaml:321]: How about
```suggestion
      any-fun: remove each item x from the beginning of a until b(x) returns false
```

**Review by cgccuser — CHANGES_REQUESTED** [review]
What emanresu said

File: documents/knowledge/elements.yaml [hunk 3]
```diff
@@ -310,14 +310,16 @@
...
+  name: Choose / random choice / set same / drop while
+  description: Binomial coefficient / choose a random items from b / same except duplicates / drop while
   arity: 2
   overloads:
       num-num: a choose b (binomial coefficient)
       num-str: choose a random items from b
       str-num: choose b random items from a
       str-str: are the set of characters in the strings the same?
+      any-fun: remove each item x from the beginning of a until b(x) returns false
+      fun-any: remove items from the beginning of b until a(x) returns false
```
Comment (cgccuser) [inline documents/knowledge/elements.yaml:322]:
```suggestion
      fun-any: remove each item x from the beginning of b until a(x) returns false
```

File: vyxal/elements.py [hunk 4]
```diff
@@ -3193,6 +3194,8 @@ def n_choose_r(lhs, rhs, ctx):
             random.choice(lhs) for _ in range(abs(int(rhs)))
         ],
         (str, str): lambda: int(set(lhs) == set(rhs)),
+        (types.FunctionType, ts[1]): lambda: n_choose_r(rhs, lhs, ctx),
```
Comment (cgccuser) [inline vyxal/elements.py:3197]: Why is this one `n_choose_r` and not `drop_while`?

**Review by cgccuser — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 4 above)
Comment (Steffan153) [reply vyxal/elements.py:3197]: `n_choose_r` is the parent function, so it doesn't really matter which, but I'll switch that around.

```suggestion
        (types.FunctionType, ts[1]): lambda: drop_while(rhs, lhs, ctx),
```

**Review by cgccuser — APPROVED** [review]
LGTM

**Review by chunkybanana — APPROVED** [review]

**Pull request merged**